plugins/l4d2/l4d2.cpp

```diff
@@ -3,176 +3,182 @@
+	escape(servername);
+	if (strcmp(servername, "") != 0) {
+		oidentity << std::endl;
+		oidentity << "\"Server name\": \"" << servername << "\","; // Set server name in identity.
+	} else {
+		oidentity << std::endl << "\"Server name\": null,";
 	}
-	
-	// context support
-	// used to disable positional audio when people are on different servers
-	_context[sizeof(_context) - 1] = '\0';
-	std::string sHost(_context);
-	// this string will change depending on your context:
-	// if in public game then server ip  -> "xxx.xxx.xxx.xxx:yyyyy"
-	// if in single player then loopback -> "loopback:0"
-	// if in menu then empty             -> ""
-	if (!sHost.empty())
-	{
-		if (sHost.find("loopback") == std::string::npos)
-		{
-			std::ostringstream newcontext;
-			newcontext << "{\"ipport\": \"" << sHost << "\"}";
-			context = newcontext.str();
-		}
-				
+
+	// Map
+	map[sizeof(map)-1] = 0; // NUL terminate queried C strings. We do this to ensure the strings from the game are NUL terminated. They should be already, but we can't take any chances.
+	escape(map);
+	if (strcmp(map, "") != 0) {
+		oidentity << std::endl;
+		oidentity << "\"Map\": \"" << map << "\","; // Set map name in identity.
+	} else {
+		oidentity << std::endl << "\"Map\": null,";
+	}
+
+	// Player
+	player[sizeof(player)-1] = 0; // NUL terminate queried C strings. We do this to ensure the strings from the game are NUL terminated. They should be already, but we can't take any chances.
+	escape(player);
+	if (strcmp(player, "") != 0) {
+		oidentity << std::endl;
+		oidentity << "\"Player\": \"" << player << "\","; // Set player nickname in identity.
+	} else {
+		oidentity << std::endl << "\"Player\": null";
+	}
+
+	oidentity << std::endl << "}";
+	identity = oidentity.str();
+	// End identity
+
+	/*
+	Mumble | Game
+	X      | X
+	Y      | Z
+	Z      | Y
+	*/
+	avatar_pos[0] = avatar_pos_corrector[0];
+	avatar_pos[1] = avatar_pos_corrector[2];
+	avatar_pos[2] = avatar_pos_corrector[1];
+
+	camera_pos[0] = camera_pos_corrector[0];
+	camera_pos[1] = camera_pos_corrector[2];
+	camera_pos[2] = camera_pos_corrector[1];
+
+	avatar_front[0] = avatar_front_corrector[0];
+	avatar_front[1] = avatar_front_corrector[2];
+	avatar_front[2] = avatar_front_corrector[1];
+
+	avatar_top[0] = avatar_top_corrector[0];
+	avatar_top[1] = avatar_top_corrector[2];
+	avatar_top[2] = avatar_top_corrector[1];
+
+	// Convert from inches to meters and sync camera vectors with avatar ones
+	for (int i=0;i<3;i++) {
+		avatar_pos[i]/=39.37f;
+		camera_pos[i]/=39.37f;
+		camera_front[i] = avatar_front[i];
+		camera_top[i] = avatar_top[i];
 	}
 
 	return true;
 }
 
+static int trylock(const std::multimap<std::wstring, unsigned long long int> &pids) {
+
+	if (! initialize(pids, L"left4dead2.exe", L"client.dll")) { // Link the game executable
+		return false;
+	}
+
+	BYTE *steamclient=getModuleAddr(L"steamclient.dll"); // Link "steamclient.dll" module
```

Comment:
You aren't linking here. The comment is misleading.
Here, you're looking up the address of steamclient.dll in memory.